src/__experimental__/components/tile/tile.spec.js

```diff
@@ -0,0 +1,185 @@
+import React from 'react';
+import TestRenderer from 'react-test-renderer';
+import 'jest-styled-components';
+import { mount } from 'enzyme';
+import { css } from 'styled-components';
+import Tile from '.';
+import { TileContent } from './tile.style';
+import Content from '../../../components/content';
+import { assertStyleMatch } from '../../../__spec_helper__/test-utils';
+
+function render(props, renderer = TestRenderer.create) {
+  return renderer(
+    <Tile { ...props }>
+      <Content key='one'>Child 1</Content>
+      <Content>Child 2</Content>
+    </Tile>
+  );
+}
+
+describe('Tile', () => {
+  it('renders base styles', () => {
+    const wrapper = render({});
+
+    expect(wrapper).toMatchSnapshot();
+  });
+
+  describe('wrapping of children in TileContent components', () => {
+    const wrapper = render({}, mount);
+    const tileContents = wrapper.find(TileContent).getElements();
+
+    it('contains one TileContent for each child', () => {
+      expect(tileContents.length).toBe(2);
+    });
+
+    it.each([0, 1])('TileContent[%i] contains the passed Content as its own child', (childIndex) => {
+      expect(tileContents[childIndex].props.children.type.name).toBe('Content');
+      expect(tileContents[childIndex].props.children.props.children).toBe(`Child ${childIndex + 1}`);
+    });
+  });
+
+  describe('styles', () => {
+    describe('as', () => {
+      it('renders a white background when as === "tile"', () => {
+        const wrapper = render({ as: 'tile' }).toJSON();
+
+        assertStyleMatch({ backgroundColor: '#FFFFFF' }, wrapper);
+      });
+
+      it('renders a transparent background when as === "transparent"', () => {
+        const wrapper = render({ as: 'transparent' }).toJSON();
+
+        assertStyleMatch({ backgroundColor: 'transparent' }, wrapper);
+      });
+    });
+
+    describe('orientation', () => {
+      describe('when it is horizontal', () => {
+        const wrapper = render({ orientation: 'horizontal' }).toJSON();
+
+        it('sets the correct flex-direction on the main wrapper', () => {
+          assertStyleMatch({ flexDirection: 'row' }, wrapper);
+        });
+
+        it('sets the TileContent display to inline', () => {
+          assertStyleMatch({ display: 'inline' }, wrapper, { modifier: css`${TileContent}` });
+        });
+
+        it('sets padding-right for all but the last TileContent', () => {
+          assertStyleMatch(
+            {
+              paddingRight: '16px'
+            },
+            wrapper,
+            { modifier: css`${`${TileContent}:not(:last-of-type)`}` }
+          );
+        });
+
+        it('sets border-left and padding-left for all but the first TileComponent', () => {
+          assertStyleMatch(
+            {
+              borderLeft: 'solid 1px #E5EAEC',
```

Comment:
It might be better to use the colours from theme object wherever possible and assert that the border matches `solid 1px ${baseTheme.tile.separator}` etc. That way if the colours are ever changed in the theme object we won't have to update all the tests

Reply:
I left it like that intentionally, as I figured the test didn't seem very useful without the value being hard-coded, as it would continue to pass regardless?

Wouldn't the point of a test like this be to ensure that if somebody is changing values in the theme, they are fully aware of the places it impacts, so you'd want this to break?

Reply:
Yeah fair point, tbh we could just use find and replace if we even need to update these, was just thinking of maintainability going forward but I'm being unnecessarily pedantic